packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -569,6 +617,23 @@ export class RouteTable extends Resource implements IRouteTable, IDependable {
 
     this.routeTableId = this.resource.attrRouteTableId;
   }
+
+  /**
+   * Adds a new custom route to the route table.
+   * @param destination The IPv4 or IPv6 CIDR block used for the destination match.
+   * @param target The gateway or endpoint targeted by the route.
+   */
+  public addRoute(id: string, destination: string, target: RouteTargetType) {
+    if (!target.gateway && !target.endpoint) {
+      throw new Error('Target is defined without a gateway or endpoint.');
+    }
```

Comment:
I feel like this check is not necessary if we tweak the implementation of `RouteTargetType` to enforce that exactly one of the two types are specified in the constructor. Something like this:

```ts
    if ((gateway && endpoint) || (!gateway && !endpoint)) {
      throw new Error("Exactly one of 'gateway' or 'endpoint' must be specified.");
    }

    if (gateway) {
      this.gateway = gateway;
    } else {
      this.endpoint = endpoint;
    }
```

Reply:
good point, validation can be in `RouteTarget`, moved it in latest rev.